File: CS/DxSample/Scheduler/CustomAppointmentStorage.cs
using DevExpress.XtraScheduler.Data;
using DevExpress.XtraScheduler;

namespace DxSample.Scheduler {
    public class CustomAppointmentStorage :AppointmentStorage {
        public CustomAppointmentStorage(SchedulerStorage storage) : base(storage) {
            SetAppointmentFactory(new CustomAppointmentFactory());
        }

        protected override DataManager<Appointment> CreateDataManager() {
            return new CustomAppointmentDataManager();
        }
    }
}

File: VB/DxSample/Scheduler/CustomAppointmentStorage.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraScheduler.Data
Imports DevExpress.XtraScheduler

Namespace DxSample.Scheduler
	Public Class CustomAppointmentStorage
		Inherits AppointmentStorage
		Public Sub New(ByVal storage As SchedulerStorage)
			MyBase.New(storage)
			SetAppointmentFactory(New CustomAppointmentFactory())
		End Sub

		Protected Overrides Function CreateDataManager() As DataManager(Of Appointment)
			Return New CustomAppointmentDataManager()
		End Function
	End Class
End Namespace
